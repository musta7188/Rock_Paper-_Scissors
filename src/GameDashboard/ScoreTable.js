import React from "react";
import { connect } from "react-redux";
import "../Style/ScoreTable.css";

export function ScoreTable({ score }) {
  return (
    <div className="score_table">
      <section className="score_table__info">
        <div>ROCK</div>
        <div>PAPER</div>
        <div>SCISSORS</div>
      </section>
      <section className="score_table__score">
        <div>SCORE</div>
        <div className="score">{score}</div>
      </section>
    </div>
  );
}

const mapStateToProps = (state) => {
  return {
    score: state.score,
  };
};

export default connect(mapStateToProps)(ScoreTable);
