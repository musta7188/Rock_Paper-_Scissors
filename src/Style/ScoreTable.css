.score_table {
  margin: 50px auto 0;
  color:white;
  display: flex;
  border: 3px solid rgba(255,255,255,.2);
	flex-wrap:wrap;
  padding:25px;
  width: 700px;
  border-radius: 10px;
  justify-content: space-between;
  max-width: 100%;
}

.score_table__info {
  font-size: 35px;
}
.score_table__info div{
  line-height: 25px;
  margin-bottom: 4px;
}
.score_table__info div:last-child{
  margin-bottom: 0px;
}

.score_table__score {
  text-align: right;
  background-color: white;
  color: #393f7b;
  text-align: center;
  width: 150px;
  padding: 20px;
  border-radius: 10px;
}

.score{
  font-size: 50px;
  font-weight: bold;
  line-height: 40px;
}


@media screen and (max-width: 375px){
  .score_table {
    margin: 30px auto 0;
    color: white;
    display: flex;
    border: 3px solid rgba(255,255,255,.2);
    flex-wrap: wrap;
    padding: 10px;
    width: 330px;
    border-radius: 10px;
    justify-content: space-between;
    max-width: 100%;
    transform: translate(6px, 10px);
  }
  
  .score_table__info {
    font-size: 20px;
    margin: 20px 18px 0px;
  }
  .score_table__info div{
    line-height: 12px;
    margin-bottom: 4px;
  }

  
  .score_table__score {
    background-color: white;
    color: #393f7b;
    text-align: center;
    width: 93px;
    height: 84px;
    padding: 10px;
    border-radius: 6px;
  }
  
  .score{
    font-size: 50px;
    font-weight: bold;
    line-height: 40px;
  }
  

}