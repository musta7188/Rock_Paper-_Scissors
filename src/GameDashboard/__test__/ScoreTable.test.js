import { render, fireEvent } from "../../test-utils";
import ScoreTable from "../ScoreTable";

describe("ScoreTable", () => {
  it("it render correctly", () => {
    const score = {
      score: 3,
    };
    const { container } = render(<ScoreTable score={score} />);
    expect(container.getElementsByClassName("score")).toHaveLength(1);
    expect(container.firstChild).toMatchSnapshot();
  });
});
