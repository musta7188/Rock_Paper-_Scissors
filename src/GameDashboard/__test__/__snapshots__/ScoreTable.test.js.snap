// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreTable it render correctly 1`] = `
<div
  class="score_table"
>
  <section
    class="score_table__info"
  >
    <div>
      ROCK
    </div>
    <div>
      PAPER
    </div>
    <div>
      SCISSORS
    </div>
  </section>
  <section
    class="score_table__score"
  >
    <div>
      SCORE
    </div>
    <div
      class="score"
    >
      0
    </div>
  </section>
</div>
`;
